### PR TITLE
Update DPS device samples .NET and SDK NuGet targets; remove TPM simulator

### PR DESCRIPTION
**Pull request by drwill-ms**
Rely on customers having HSMs which are common enough these days.

### CODE REVIEW

File: provisioning/Samples/device/TpmSample/ProvisioningDeviceClientSample.cs [hunk 1]
```diff
@@ -27,64 +26,41 @@ public ProvisioningDeviceClientSample(Parameters parameters)
...
-                    SecurityProviderTpmSimulator.StopSimulatorProcess();
-                }
+            Console.WriteLine($"Testing the provisioned device with IoT Hub...");
+            using DeviceClient iotClient = DeviceClient.Create(result.AssignedHub, auth, _parameters.TransportType);
 
-                security?.Dispose();
-            }
+            Console.WriteLine("Sending a telemetry message...");
+            using var message = new Message(Encoding.UTF8.GetBytes("TestMessage"));
+            await iotClient.SendEventAsync(message);
```
Comment (timtay-microsoft) [inline provisioning/Samples/device/TpmSample/ProvisioningDeviceClientSample.cs:63]: Should we close this client before it gets disposed?

**Review by timtay-microsoft — APPROVED** [review]

**Review by abhipsaMisra — APPROVED** [review]

**Review by azabbasi — APPROVED** [review]

**Pull request merged**